cfme/scripting/bz.py

```diff
@@ -0,0 +1,221 @@
+                qe_test_coverage = flag["status"]
+                break
+        # append the BZ
+        bz_list.append(BZTestCoverage(id=bug_id, qe_test_coverage=qe_test_coverage))
+
+    return bz_list
+
+
+def cleanup():
+    click.echo("Removing the BZ report file, bz-report.yaml")
+    try:
+        os.remove("bz-report.yaml")
+    except OSError:
+        logger.exception("bz-report.yaml not found")
+
+
+@click.group(help="Functions for generating reports on BZs included in test suite metadata")
+def main():
+    pass
+
+
+@main.command(help="Generate BZ report on BZs that have coverage given a directory")
+@click.argument("directory", default="cfme/tests/")
+def report(directory):
+    get_report(directory)
+
+
+@main.command(help="List open/closed BZs that have test coverage")
+@click.argument("directory", default="cfme/tests/")
+@click.option(
+    "--all",
+    "-a",
+    "bz_status",
+    is_flag=True,
+    help="list all BZs with coverage",
+    default=True,
+    show_default=True,
+    flag_value="all_bzs",
+)
+@click.option(
+    "--open",
+    "-o",
+    "bz_status",
+    is_flag=True,
+    help="list open BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="open_bzs"
+)
+@click.option(
+    "--closed",
+    "-c",
+    "bz_status",
+    is_flag=True,
+    help="list closed BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="closed_bzs"
+)
+def list(directory, bz_status):
+    info = get_report(directory)
+
+    # get dict of information
+    status = STATUS[bz_status]
+
+    if status["val"]:
+        bz_list = [bug_id for bug_id in info.keys() if info[bug_id]["is_open"] == status["val"]]
+    else:
+        bz_list = [bug_id for bug_id in info.keys()]
+
+    if bz_list:
+        click.echo("The following BZ's {}: \n{}".format(status["text"], "\n".join(bz_list)))
+    else:
+        click.echo("I found no BZ's that {} BZ's".format(status["text"]))
+    cleanup()
+
+
+@main.command(help="Set QE test coverage flag based on automates/coverage metadata")
+@click.argument("directory", default="cfme/tests")
+@click.option(
+    "-s",
+    "--set",
+    "set_bzs",
+    is_flag=True,
+    help="Set QE test coverage on BZs that are marked in coverage/auotmates test metadata",
+    default=False,
+    show_default=True
+)
+@click.option(
+    "--all",
+    "-a",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on all BZs",
+    default=True,
+    show_default=True,
+    flag_value="all_bzs",
+)
+@click.option(
+    "--open",
+    "-o",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on only open BZs",
+    default=False,
+    show_default=True,
+    flag_value="open_bzs"
+)
+@click.option(
+    "--closed",
+    "-c",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on only closed BZs",
+    default=False,
+    show_default=True,
+    flag_value="closed_bzs"
+)
+def coverage(directory, set_bzs, bz_status):
```

Comment:
In running `miq bz coverage`, I'm seeing some misleading/conflicting information about the BZ's I can/should update.

```
The following BZs should have qe_test_coverage set to '+': 
    id: 1443411, qe_test_coverage: +
```

As shown in the output, 1443411 has qe_test_coverage set to `+`.  Turns out there is a clone of this BZ that has qe_test_coverage set to `-`.

We need to either not take BZ clones into effect when checking/setting coverage, or we need to identify and include them in the output instead of the marked BZ number.

Reply:
As I've written it, this does nothing with clones at the moment. Perhaps this could be a future enhancement. I think this is just a "doh" moment where I forgot to check if the BZs had their coverage set to "+" in the coverage audit script (`get_qe_test_coverage`). I added a condition in there to prevent this from happening.